deployment/cre/ocr3/v2/changeset/sequences/deploy_ocr3.go

```diff
@@ -0,0 +1,127 @@
+	RegistryChainSel uint64
+	Qualifier        string
+
+	DONs         []contracts.ConfigureCREDON
+	OracleConfig *ocr3.OracleConfig
+	DryRun       bool
+
+	MCMSConfig *ocr3.MCMSConfig
+}
+
+func (c DeployOCR3Input) Validate() error {
+	return nil
+}
+
+type DeployOCR3Output struct {
+	ChainSelector uint64
+	Address       string
+	Type          string
+	Version       string
+	Labels        []string
+}
+
+var DeployOCR3 = operations.NewSequence(
+	"deploy-ocr3",
+	semver.MustParse("1.0.0"),
+	"Deploys the OCR3 contract",
+	func(b operations.Bundle, deps DeployOCR3Deps, input DeployOCR3Input) (DeployOCR3Output, error) {
+		// Set default qualifier if not provided
+		qualifier := input.Qualifier
+		if qualifier == "" {
+			qualifier = "capability_consensus"
+		}
+
+		// Step 1: Deploy OCR3 Contract for Consensus Capability
+		ocr3DeploymentReport, err := operations.ExecuteOperation(b, contracts.DeployOCR3, contracts.DeployOCR3Deps{Env: deps.Env}, contracts.DeployOCR3Input{
+			ChainSelector: input.RegistryChainSel,
+			Qualifier:     qualifier,
+		})
+		if err != nil {
+			return DeployOCR3Output{}, err
+		}
+
+		ocr3ContractAddress := common.HexToAddress(ocr3DeploymentReport.Output.Address)
+
+		// Update the environment datastore to include the newly deployed OCR3 contract
+		deps.Env.DataStore = ocr3DeploymentReport.Output.Datastore
+
+		// Step 2: Get the capabilities registry contract
+		capabilitiesRegistry, err := getCapabilitiesRegistryContract(deps, input)
+		if err != nil {
+			return DeployOCR3Output{}, fmt.Errorf("failed to get capabilities registry: %w", err)
+		}
+
+		// Step 3: Configure OCR3 Contract with DONs
+		deps.Env.Logger.Infow("Configuring OCR3 contract with DONs",
+			"numDONs", len(input.DONs),
+			"dryRun", input.DryRun)
+
+		_, err = operations.ExecuteOperation(b, contracts.ConfigureOCR3, contracts.ConfigureOCR3Deps{
+			Env:                  deps.Env,
+			WriteGeneratedConfig: deps.WriteGeneratedConfig,
+			Registry:             capabilitiesRegistry,
+		}, contracts.ConfigureOCR3Input{
+			ContractAddress:  &ocr3ContractAddress,
+			RegistryChainSel: input.RegistryChainSel,
+			DONs:             input.DONs,
+			Config:           input.OracleConfig,
+			DryRun:           input.DryRun,
+			MCMSConfig:       input.MCMSConfig,
+		})
+		if err != nil {
+			return DeployOCR3Output{}, fmt.Errorf("failed to configure OCR3 contract: %w", err)
+		}
+
+		return DeployOCR3Output{
+			ChainSelector: ocr3DeploymentReport.Output.ChainSelector,
+			Address:       ocr3DeploymentReport.Output.Address,
+			Type:          ocr3DeploymentReport.Output.Type,
+			Version:       ocr3DeploymentReport.Output.Version,
+			Labels:        ocr3DeploymentReport.Output.Labels,
+		}, nil
+	},
+)
+
+func getCapabilitiesRegistryContract(deps DeployOCR3Deps, input DeployOCR3Input) (*capabilities_registry_v2.CapabilitiesRegistry, error) {
+	refs := deps.Env.DataStore.Addresses().Filter(datastore.AddressRefByType("CapabilitiesRegistry"))
```

Comment:
I would also add a filter by chain selector, to make sure the item you are then using is the one from the chain you are working on:

```
datastore.AddressRefByChainSelector(input.RegistryChainSel)
```